airbyte/_executor.py

```diff
@@ -450,3 +451,55 @@ def uninstall(self) -> NoReturn:
     def execute(self, args: list[str]) -> Iterator[str]:
         with _stream_from_subprocess([str(self.path), *args]) as stream:
             yield from stream
+
+
+class DockerExecutor(Executor):
+    def __init__(
+        self,
+        name: str | None = None,
+        *,
+        executable: list[str],
+        target_version: str | None = None,
+    ) -> None:
```

Comment:
See notes in other comments re: suggestion to take in docker image (and tag?) along with optional temp dir(s).